src/passes/OptimizeInstructions.cpp

```diff
@@ -994,6 +994,20 @@ struct OptimizeInstructions
           //   return binary;
           // }
         }
+      } else if (binary->op == RemSInt32) {
+        // bool((signed)x % C_pot)  ==>  bool(x & (C_pot - 1))
```

Comment:
The case of `isSignedMin` is not documented in the comment here?

About that case of signed min, isn't it true that
```
(signed)x % -2147483648  ==>  x == -2147483648
```
even outside a boolean context?

Reply:
> About that case of signed min, isn't it true that
> ```
> (signed)x % -2147483648  ==>  x == -2147483648
> ```

No, let's see how LLVM handle this:
https://godbolt.org/z/K51Knj

So inside boolean context it `(signed)x % min` -> `(signed)x & max`

Reply:
Oh, right, I was thinking of division...

Reply:
Ah, sorry. Forget about this. Done!